x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/operator/TopNOperator.java

```diff
@@ -22,12 +24,47 @@
 
 import java.util.ArrayList;
 import java.util.Arrays;
+import java.util.BitSet;
 import java.util.Collections;
 import java.util.Iterator;
 import java.util.List;
+import java.util.function.BiConsumer;
 
 public class TopNOperator implements Operator {
 
+    private static final byte SEPARATOR = 0x0; // separator for values inside the BytesRef sorting key
+    private static final byte SMALL_NULL = 0x01; // "null" representation for "nulls first"
+    private static final byte BIG_NULL = 0x02; // "null" representation for "nulls last"
+
+    public enum Encoder {
+        IP_BYTES_REF_ENCODER((rowValue, bytesRefBuilder) -> bytesRefBuilder.append(rowValue)),
+        BYTES_REF_ENCODER((rowValue, bytesRefBuilder) -> {
+            // add one bit to every byte so that there are no "0" bytes in the provided bytes. The only "0" bytes are
+            // those defined as separators
+            int end = rowValue.offset + rowValue.length;
+            for (int i = rowValue.offset; i < end; i++) {
+                byte b = rowValue.bytes[i];
+                if ((b & 0b1000_0000) == 0) {
+                    b++;
+                }
```

Comment:
To avoid the conditional, since you don't care about the intermediate bits you can flip over the most
significant bit (MSB) check into a least significant bit (LSB) by shifting its position and doing an and.
The block can be replaced with `b += (b >> 7) & 1`